### PR TITLE
Vaxrank report changes

### LINKED ISSUE #58: Add CSV report(s)
CSV per variants w/ N vaccine peptides in it

### PR DESCRIPTION
**Pull request by julia326**
- added manufacturer report gen and consolidated with existing `--output-csv` functionality
what's getting saved now is the ranked variant list output of core_logic.py, rather than processed template data
- a bit of manufacturability tuple creation refactoring to make it clearer (from Alex)
- refactored cli.py to make it easier/cleaner to add new non-template report outputs
- couple logging changes:
  - added a file log to default logging config (helps with grepping/debugging locally)
  - logging line numbers
- got rid of pickling in favor of JSON (using `serializable`)
- updating `shellinford` to py3-friendly version
- in the HTML report, moving the note about how this is loaded from a JSON file to a separate ilne outside the args list

Fixes #58, #68, #69

### CODE REVIEW

File: vaxrank/cli.py [hunk 1]
```diff
@@ -190,38 +217,36 @@ def load_template_data(args):
...
-        'patient_id': args.output_patient_id,
-        'final_review': args.output_final_review,
-        'reviewers': args.output_reviewed_by.split(','),
-    }
-
-    template_data_creator = TemplateDataCreator(
-        ranked_variants_with_vaccine_peptides=ranked_list_for_report,
-        mhc_alleles=mhc_alleles,
-        variants=variants,
+    # TODO(julia): is this right?
```
Comment (julia326) [inline vaxrank/cli.py]: @iskandr is this reasonable?

File: vaxrank/report.py [hunk 2]
```diff
@@ -197,11 +216,14 @@ def compute_template_data(self):
...
             variant_short_description = variant.short_description
             if len(vaccine_peptides) == 0:
-                logger.info("Skipping %s, no vaccine peptides", variant_short_description)
+                logger.info("Skipping gene(s) %s, variant %s: no vaccine peptides",
+                    variant.gene_names, variant_short_description)
                 continue
 
             variant_data = self._variant_data(vaccine_peptides[0])
-            predicted_effect = self.variants_to_top_coding_effect_dict.get(variant)
+            # TODO(julia): is this right?
```
Comment (julia326) [inline vaxrank/report.py]: @iskandr you recently added `supporting_reference_transcripts` as a property on the MutantProteinFragment, which is connected to a particular peptide. but when we display a predicted effect, it's variant-level as opposed to peptide-level. what's the best thing to do here? do you take the top peptide and grab its supporting transcripts? or do what i'm doing here now?

File: vaxrank/cli.py (hunk 1 above)
Comment (iskandr) [inline vaxrank/cli.py]: Is that last `items()` necessary?

Comment (julia326) [inline vaxrank/cli.py]: nope

**Comment by ihodes** [comment]
LGTM!

File: vaxrank/report.py [hunk 3]
```diff
@@ -197,11 +216,16 @@ def compute_template_data(self):
...
+                logger.info("Skipping gene(s) %s, variant %s: no vaccine peptides",
+                    variant.gene_names, variant_short_description)
                 continue
 
-            variant_data = self._variant_data(vaccine_peptides[0])
-            predicted_effect = self.variants_to_top_coding_effect_dict.get(variant)
+            top_peptide = vaccine_peptides[0]
+            variant_data = self._variant_data(top_peptide)
+            # TODO(julia): is this right?
+            effects = [variant.effect_on_transcript(t) for t in
```
Comment (iskandr) [inline vaxrank/report.py]: LGTM

**Comment by ihodes** [comment]
LGTM!

**Pull request merged**